ibis/backends/sql/__init__.py

```diff
@@ -450,7 +450,7 @@ def insert(
             target=name, source=obj, db=db, catalog=catalog
         )
 
-        with self._safe_raw_sql(query):
+        with self._safe_raw_sql(query.sql(self.dialect)):
```

Comment:
`_safe_raw_sql` wraps `raw_sql` which accepts either a string or a `sqlglot.Expression` so this compilation isn't needed